### PR TITLE
New Functionality to arrays - arrayRemove

### PR DESCRIPTION
**Pull request by rajatmohan22**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

A new function on arrays called ArrayRemove. Its working is as follows:` arayRemove(x -> cond, arr)`, This removes those elements in the array for which the condition is true.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

**Motivation:** This function aims to address certain confusions caused by arrayFilter. The arrayFilter function operates mysteriously using the syntax` arrayRemove(x -> cond, arr)`, where the elements that satisfy the condition are retained in the array. This can lead to confusion for users. To simplify and clarify the process, we introduce the arrayRemove function, which works directly as its name suggests. It removes the elements that pass the given condition, offering a more intuitive and straightforward approach.

**parameters:** : 
-> arrayRemove(arr, elem).
arr refers to the array and elem refers to the element to be removed.

-> Works with lambda functions as well:
arrayRemove(arr, x-> cond)
arr refers to the array. the second parameter refers to the condition by which the element must be removed.

**Example:**

 SELECT 
    id,
    ARRAY REMOVE(numbers, x -> x % 2 = 0) AS new_numbers
FROM my_table;

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit a205860b36c38cb1b03c6e8cd75f9300f48bb852 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52371/a205860b36c38cb1b03c6e8cd75f9300f48bb852/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

**Comment by rajatmohan22** [comment]
@alexey-milovidov Could you please help me out? My code definitely adhers to the stlyle guide at https://clickhouse.com/docs/en/development/style.
why is it failing ? Your help would be truly appreciated !

**Comment by alexey-milovidov** [comment]
Please don't remove comments. There was a problem with trailing whitespace and with one extra beginning whitespace.
You can set up your IDE (CLion, VSCode, KDevelop, QTCreator, vim, emacs) to remove trailing whitespace on save.

**Comment by rajatmohan22** [comment]
> 

Will do that (I removed them because the Style Check was failing multiple times. I did my best to keep the comments and still pass them) and update the branch

Is there anything else I am supposed to do? the query optimization tests is failing.

**Comment by rajatmohan22** [comment]
also should i update docs/en/sql-reference/functions/array-functions.md ? When i update it, the style check is failing

**Comment by rajatmohan22** [comment]
@alexey-milovidov @mindreframer Could you please check if the commit can be merged? the overall status of the commit is a success

File: docs/en/sql-reference/functions/array-functions.md [hunk 1]
```diff
@@ -1794,6 +1794,52 @@ Return value type is always [Float64](../../sql-reference/data-types/float.md).
 │ 6   │ Float64                                                                                  │
 └─────┴──────────────────────────────────────────────────────────────────────────────────────────┘
 ```
+## array
```
Comment (evillique) [inline docs/en/sql-reference/functions/array-functions.md:1797]:
```suggestion
## arrayRemove
```

File: tests/queries/0_stateless/02832_array_remove.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+SELECT sum(length(arr)) FROM (SELECT arrayMap(x -> toString(x), range(number % 10)) AS arr FROM (SELECT * FROM system.numbers LIMIT 1000));
```
Comment (evillique) [inline tests/queries/0_stateless/02832_array_remove.sql:1]: Am I missing something or does the test not contain `arrayRemove` function?

File: src/Functions/array/arrayRemove.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include "FunctionArrayMapped.h"
+
+
+namespace DB
+{/** arrayRemove(x -> predicate, array) - Remove from the array only the elements for which the expression is true.
```
Comment (evillique) [inline src/Functions/array/arrayRemove.h:9]: The definition does not match the description in the docs

**Comment by alexey-milovidov** [comment]
@rajatmohan22 Why closed? We need it...

**Comment by rajatmohan22** [comment]
@alexey-milovidov I was having a lot of issues building the project and testing locally. I'll try rebuilding it and submit a better PR asap

**Comment by rajatmohan22** [comment]
@alexey-milovidov I need help building the project on my local system. Is there a mailing list to join please? I'll be able to contribute valuable work i promise.

**Comment by alexey-milovidov** [comment]
https://clickhouse.com/slack

**Pull request closed without merging**